Python/Product/Wsl/Debugger/WslLauncher.cs

```diff
@@ -0,0 +1,327 @@
+            }
+
+            return (
+                "if hash python{0}.{1} 2>/dev/null; then export py=python{0}.{1};" +
+                "elif hash python{0} 2>/dev/null; then export py=python{0}; echo 'WARNING: Using python{0} instead of python{0}.{1}';" +
+                "else echo 'Cannot locate Python {0}.{1}'; read -p 'Press Enter to exit . . .'; exit 1;" +
+                "fi"
+                ).FormatInvariant(version.Major, version.Minor);
+        }
+
+        private string CreateScriptFile(params string[] lines) {
+            var sh = Path.GetTempFileName();
+
+            var content = string.Join("\n", lines.Where(l => !string.IsNullOrEmpty(l)));
+#if !DEBUG
+            content += "\nrm -f " + QuoteSingleArgument(FixPath(sh)) + "\n";
+#endif
+
+            var bytes = new UTF8Encoding(false).GetBytes(content);
+            using (var stream = new FileStream(sh, FileMode.Truncate, FileAccess.Write, FileShare.None)) {
+                stream.Write(bytes, 0, bytes.Length);
+                stream.Flush();
+            }
+
+            return sh;
+        }
+
+        private Process StartWithoutDebugger(LaunchConfiguration config) {
+            var sh = CreateScriptFile(
+                GetPyVariable(config.Interpreter?.Version),
+                "$py {0} {1} {2}".FormatInvariant(
+                    config.InterpreterArguments,
+                    config.ScriptName == null ? "" : QuoteSingleArgument(FixPath(config.ScriptName)),
+                    config.ScriptArguments
+                ),
+                "read -p 'Press Enter to exit . . .'"
+            );
+            
+            var psi = new ProcessStartInfo {
+                FileName = _bashPath.Value ?? throw new InvalidOperationException("WSL is not installed"),
+                Arguments = QuoteSingleArgument(FixPath(sh)),
+                WorkingDirectory = config.WorkingDirectory,
+                UseShellExecute = false
+            };
+
+            return Process.Start(psi);
+        }
+
+        private static string PtvsdSearchPath {
+            get {
+                return PathUtils.GetParent(PathUtils.GetParent(PythonToolsInstallPath.GetFile("ptvsd\\__main__.py")));
+            }
+        }
+
+        private void StartWithDebugger(LaunchConfiguration config) {
+            if (string.IsNullOrEmpty(config.ScriptName)) {
+                throw new ArgumentException("script name");
+            }
+
+            int port = Enumerable.Range(new Random().Next(49152, 65536), 60000)
+                .Except(IPGlobalProperties.GetIPGlobalProperties().GetActiveTcpConnections().Select(c => c.LocalEndPoint.Port))
+                .First();
+
+            string secret = Path.GetRandomFileName().Replace(".", "");
+
+            var sh = CreateScriptFile(
+                GetPyVariable(config.Interpreter?.Version),
+                string.Join(" ",
+                    "export PYTHONPATH={0};".FormatInvariant(QuoteSingleArgument(FixPath(PtvsdSearchPath))),
```

Comment:
Yes, though we still need to provide search paths here. Separating it into its own line in the script entirely will make life much easier for debugging.